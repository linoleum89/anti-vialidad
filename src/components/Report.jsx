import React from "react";
import PropTypes from 'prop-types';
import { Card, Button, Col } from "react-bootstrap";

const Report = props => {
  return (
    <Col md="auto">
      <Card className={props.className}>
        <Card.Body>
          <Card.Title>{props.name}</Card.Title>
          <Card.Text>{props.description}</Card.Text>
          {props.coordinates && props.coordinates.length > 0 && (
            <Button coordinates={props.coordinates} variant="primary">
              Open map
            </Button>
          )}
        </Card.Body>
      </Card>
    </Col>
  );
};

Report.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
  description: PropTypes.string,
  coordinates: PropTypes.array
} 

export default Report;
