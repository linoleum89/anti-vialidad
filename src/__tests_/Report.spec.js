import React from "react";
import TestRenderer from "react-test-renderer";
import Report from "../components/Report";

describe("Report Component", () => {
  let component,
    tree,
    props = {
      className: "red",
      name: "some report name",
      description: "some description",
      coordinates: []
    };
  beforeAll(() => {
    component = TestRenderer.create(<Report {...props} />);
    tree = component.toJSON();
  });
  it("should match snapshot", () => {
    expect(tree).toMatchSnapshot();
  });
  it("should match the class name", () => {
    expect(tree.children[0].props.className).toEqual("red card");
  });
});
