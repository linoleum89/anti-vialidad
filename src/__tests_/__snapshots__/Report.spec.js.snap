// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report Component should match snapshot 1`] = `
<div
  className="col-md-auto"
>
  <div
    className="red card"
  >
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        some report name
      </div>
      <p
        className="card-text"
      >
        some description
      </p>
    </div>
  </div>
</div>
`;
